src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,192 @@
+                        // is a vector containing as many lanes as the last argument and the vector
+                        // begins at lane number specified by the penultimate arg.
+                        // So, slice_vector(2, a, b, 126, 128) gives a vector of 128 lanes starting
+                        // from lanes 126 through 253 of the concatenated vector a.b.
+                        debug(4) << "HexagonAlignLoads: Unaligned Load: Converting " << (Expr) op << " into ...\n";
+                        expr = Call::make(op->type, Call::slice_vector, { make_two(Int(32)), load_low, load_high, make_const(Int(32), (base_mod + rem_mod)),
+                                    make_const(Int(32), lanes) }, Call::PureIntrinsic);
+                        debug(4) <<  "... " << expr << "\n";
+                        return;
+                    }
+                } else if (ramp && stride && stride->value == 2) {
+                    // We'll try to break this into two dense loads followed by a shuffle.
+                    Expr base_a = ramp->base, base_b = ramp->base + ramp->lanes;
+                    bool shifted_b = false;
+                    int lanes = ramp->lanes;
+
+                    if (op->param.defined()) {
+                        // We need to be able to figure out if buffer_base + base_a is aligned. If not,
+                        // we may have to shift base_b left by one so as to not read beyond the end
+                        // of an external buffer.
+                        ModulusRemainder mod_rem = get_alignment_info(base_a);
+                        int base_lanes_off = mod_imp(op->param.host_alignment(), lanes) +
+                            mod_imp(mod_rem.modulus, lanes);
+                        int rem_mod = mod_imp(mod_rem.remainder, lanes);
+                        if (base_lanes_off + rem_mod) {
+                            debug(4) << "HexagonAlignLoads: base_a is unaligned: shifting base_b\n";
+                            debug(4) << "HexagonAlignLoads: Type: " << op->type << "\n";
+                            debug(4) << "HexagonAlignLoads: Index: " << index << "\n";
+                            base_b -= 1;
+                            shifted_b = true;
+                        }
+                    }
+
+                    Expr ramp_a = Ramp::make(base_a, 1, lanes);
+                    Expr ramp_b = Ramp::make(base_b, 1, lanes);
+                    Expr vec_a = mutate(Load::make(op->type, op->name, ramp_a, op->image, op->param));
+                    Expr vec_b = mutate(Load::make(op->type, op->name, ramp_b, op->image, op->param));
+                    Expr dbl_vec = Call::make(op->type.with_lanes(lanes*2), Call::concat_vectors, { vec_a, vec_b }, Call::PureIntrinsic);
+
+                    std::vector<Expr> args;
+                    args.push_back(dbl_vec);
+                    for (int i = 0; i < lanes/2; ++i) {
+                        args.push_back(make_const(Int(32), i*2));
```

Comment:
I'd just push back the integer. Expr constructs from int the way you would want.